### ISSUE TITLE
Odd looking button on logout menu [Xfce]

### ISSUE THREAD

**Issue by mehedirm6244** [lah7/gtk3-classic#38]
At first I thought it to be an issue of the GTK theme I was using, but it isn't. Without this patch, the "Cancel" button looks how it's supposed to look. But with this patch:
![image](https://user-images.githubusercontent.com/86041547/131506133-e1875a3d-f9fd-4b6c-9215-204a3d4a8a6e.png)
Is the button even connected to anything of CSD?

### Environment
DE: Xfce 4.16
Distro: Arch
Using `gtk3-classic 3.24.30-1` and `lib32-gtk3-classic 3.24.30-1`

**Comment by fulalas** [comment]
I can confirm this issue on Slackware x64 current.

**Comment by lah7** [comment]
This issue is a few years old - is it still an issue with the latest XFCE?

**Comment by fulalas** [comment]
Yes. Using Xfce 4.20 + gtk3-classic built from master a few days ago I can still see some misalignment of the cancel button.

**Comment by mehedirm6244** [comment]
Gotta look into the source code of the of the logout modal. I guess there's some wacky things going on with the cancel button.

**Comment by lah7** [comment]
Since it only happens under `gtk3-classic`, one could try excluding patches to identify which one is causing the issue.

My guess might be [fixes__labels-wrapping.patch](https://github.com/lah7/gtk3-classic/blob/master/fixes__labels-wrapping.patch)

**Comment by fulalas** [comment]
I just compiled GTK+3 with and without classic patches. Here's the result:

Original GTK+3 **without** classic patches:
![Image](https://github.com/user-attachments/assets/e80ae8fb-604f-475c-ac0e-1ba3af54e4d9)

GTK+3 with classic patches:
![Image](https://github.com/user-attachments/assets/a7940ad2-f56c-45db-be51-7ab3e286f2c8)

It's been a looong time since I compiled GTK+3 without classic patches and I didn't remember how this screen looked like without them. It seems the only 2 differences is that the dialog is a bit smaller and the cancel button is closer to the boarders, which I personally think it's uglier than when using classic patches.

P.S.: I'm assuming I don't need to recompile Xfce for this test.

**Comment by lah7** [comment]
@fulalas Correct, XFCE doesn't need recompiling, since GTK 3 is being used a linked library. From the screenshot, it looks like the "Cancel" button looks presentable now. Was your [earlier comment](https://github.com/lah7/gtk3-classic/issues/38#issuecomment-2806688567) confirming the misalignment as bad as @mehedirm6244's [screenshot](https://github.com/lah7/gtk3-classic/issues/38#issue-983830253)? Or is it 'fixed' for you after GTK 3 recompiled?

@mehedirm6244 I did notice a withdrawn comment from a day ago. Is it still a problem on your end?

For a way forward, I'm not sure what we can do unless the specific patch is identified. Even then, I'm just here to ensure patches build rather then write them, so this issue would continue to remain open for anyone to pick up if they're motivated for improving the patch or find a solution (with CSS perhaps?). Or, if newer version of XFCE has 'fixed' this, perhaps we can close the issue?

**Comment by fulalas** [comment]
> Was your https://github.com/lah7/gtk3-classic/issues/38#issuecomment-2806688567 confirming the misalignment as bad as @mehedirm6244's https://github.com/lah7/gtk3-classic/issues/38#issue-983830253?

I don't remember, to be honest.

In my opinion, this is super minor, and the current combination of GTK+3 and classic is better than the original 'pure' GTK+3, so I wouldn't mind closing this.
